x-pack/platform/plugins/shared/ml/public/application/components/job_selector/group_or_job_selector_menu/job_selector_button.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useMemo, useState } from 'react';
+import type { EuiContextMenuPanelDescriptor } from '@elastic/eui';
+import { EuiButton, EuiContextMenu, EuiPopover, useGeneratedHtmlId } from '@elastic/eui';
+import { useUrlState } from '@kbn/ml-url-state';
+import type { MlPages } from '../../../../../common/constants/locator';
+import { useJobInfoFlyouts } from '../../../jobs/components/job_details_flyout';
+import { useMlKibana } from '../../../contexts/kibana';
+import { getOptionsForJobSelectorMenuItems } from './get_options_for_job_selector_menu';
+
+interface Props {
+  jobId: string;
+  page: MlPages;
+  onRemoveJobId: (jobOrGroupId: string[]) => void;
+  removeJobIdDisabled: boolean;
+}
+
+export const AnomalyDetectionInfoButton: FC<Props> = ({
+  jobId,
+  page,
+  onRemoveJobId,
+  removeJobIdDisabled,
+}) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const {
+    services: {
+      share,
+      application: { navigateToUrl },
+    },
+  } = useMlKibana();
+  const [globalState] = useUrlState('_g');
+
+  const popoverId = useGeneratedHtmlId({
+    prefix: 'adJobInfoContextMenu',
+    suffix: jobId,
+  });
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const { setActiveFlyout, setActiveJobId } = useJobInfoFlyouts();
+  const panels = useMemo(
+    () => {
+      return [
+        {
+          id: 0,
+          items: getOptionsForJobSelectorMenuItems({
+            jobId,
+            page,
+            onRemoveJobId,
+            removeJobIdDisabled,
+            showRemoveJobId: true,
+            closePopover,
+            globalState,
+            setActiveFlyout,
+            setActiveJobId,
+            navigateToUrl,
+            share,
+          }),
+        },
+      ] as EuiContextMenuPanelDescriptor[];
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Super small nit: could you add what you're omitting? Would be good to have context when looking at this as to why something is left out of the deps array.